src/Adapter/AbstractResourceAdapter.php

```diff
@@ -228,4 +223,22 @@ protected function fillRelated($record, ResourceObject $resource, EncodingParame
         // no-op
     }
 
+    /**
+     * @param mixed $record
+     * @param ResourceObject $resource
+     * @param EncodingParametersInterface $parameters
+     * @return AsynchronousProcess|mixed
+     */
+    protected function fillAndPersist($record, ResourceObject $resource, EncodingParametersInterface $parameters)
+    {
+        $this->fill($record, $resource, $parameters);
+
+        if ($async = $this->persist($record)) {
```

Comment:
For better backwards compatibility this needs to check whether it is an instance of AsynchronousProcess before returning.